voluncheer/opportunityboard/views/postanopportunity.py

```diff
@@ -2,22 +2,50 @@
 from django.shortcuts import render
 
 from opportunityboard.forms.postanopportunity import PostAnOpportunityForm
+from profiles.models import Organization
 
 
 def post_an_opportunity(request):
     """create a new Opportunity and save it to the database."""
+    user = request.user
+    if user.is_anonymous:
+        return redirect("home")
+    if user.is_organization:
+        organization_profile = Organization.objects.get(pk=user)
     if request.method == "POST":
-        form = PostAnOpportunityForm(request.POST, instance=request.user)
+        form = PostAnOpportunityForm(request.POST, request.FILES, instance=request.user)
         if form.is_valid():
             form.save()
             return redirect("home")
```

Comment:
We should have an `else `branch here in case the form is not valid. I'm not sure if that is just a matter of raising an error or perhaps displaying an error message on the page.